src/internal/packager/sbom/viewer/compare.js

```diff
@@ -0,0 +1,173 @@
+const sbomSelector = document.getElementById("sbom-selector")
+const distroInfo = document.getElementById("distro-info")
+const modal = document.getElementById("modal")
+const modalFader = document.getElementById("modal-fader")
+const modalTitle = document.getElementById("modal-title")
+const modalContent = document.getElementById("modal-content")
+const artifactsTable = document.createElement("table")
+
+document.body.appendChild(artifactsTable);
+
+function initSelector() {
+    sbomSelector.add(new Option("-", "-", true, true));
+
+    ZARF_SBOM_LIST.sort().forEach(item => {
+        sbomSelector.add(new Option(item, item, false, false));
+    });
+}
+
+function compare() {
+    if (document.getElementById("leftJson").files.length == 0 || document.getElementById("rightJson").files.length == 0) {
+        showModal("Unable to Compare", "You must select 2 files from the file browsers")
+        return
+    }
+
+    let leftJson = document.getElementById("leftJson").files[0];
+    let rightJson = document.getElementById("rightJson").files[0];
+
+    let leftReader = new FileReader();
+    leftReader.readAsText(leftJson);
+
+    leftReader.onload = function() {
+        try {
+            let leftData = JSON.parse(leftReader.result);
+            const leftMap = {}
+            leftData.artifacts.map(artifact => {
+                if (!leftMap[artifact.name]) { leftMap[artifact.name] = {} }
```

Comment:
one-lining these probable aren't great, some other possible options (though I'm not certain what the JS engine efficiency is in variable assignment if re-assigning)

```javascript

// Same code, just wihtout linebreak
if (!leftMap[artifact.name]) {
  leftMap[artifact.name] = {};
}

// Same but shorten prop
let name = artifact.name
if (!leftMap[name]) {
  leftMap[name] = {};
}

// One-liner, but variable re-assingment
leftMap[artifact.name] = leftMap[artifact.name] || {};

// Use a helper function
ensure(leftMap, artifact.name);
function ensure(map, el) {
  if (!map[el]) {
    map[el] = {};
  }
}

```